src/lib.rs

```diff
@@ -337,6 +338,12 @@ impl Parser {
                     // Continue collecting bytes into param
                     self.param *= 10;
                     self.param += (byte - b'0') as i64;
+
+                    // Mimic XTerm's overflow-prevention behavior
+                    if self.param > u16::MAX as i64 {
```

Comment:
I thought about changing `self.param`, but I wasn't sure if there was a specific reason that you exposed the params as `i64`s to consumer of the library.

I couldn't find any documentation as to the maximum value allowed for CSI/DCS parameters. Xterm's behavior is to [limit it to u16::MAX](https://github.com/joejulian/xterm/blob/defc6dd5684a12dc8e56cb6973ef973e7a32caa3/charproc.c#L2216-L2217), so I chose that for compatibility with the de-facto spec. However, I can change it if you'd like.